command/agent/retry_join.go

```diff
@@ -4,14 +4,28 @@
 package agent
 
 import (
+	"context"
 	"fmt"
-	golog "log"
+	"net"
+	"regexp"
 	"strings"
 	"time"
 
+	golog "log"
+
+	hclog "github.com/hashicorp/go-hclog"
 	log "github.com/hashicorp/go-hclog"
```

Comment:
This is going to sound nitpicky (:grinning:): but we've got two different named imports of `go-hclog` here. Let's pick one and stick with it. Also, let's keep the `golog "log"` import with the group of stdlib imports above.